### PR TITLE
Assorted styling/layout improvements

### LINKED ISSUE #2126: Mobile dropdown menu (from hamburger icon) apps & grains are missing icons
Additionally,

**Mobile dropdown menu (from hamburger icon) blocks the entire screen**
- navigation becomes more difficult if you can't remember/see what your current UI view is

Current screenshot (Nexus 5):
image

I've also been wondering why we haven't aren't displaying the profile pictures for the user in the top bar or hamburger menu.

Suggested UI:

image

Although eventually I we may want to re-order the items in the list and style grains, apps, and account differently.

### LINKED ISSUE #2139: Apps & grains(mobile): search box displays oddly in the top right
Current screenshots (nexus 5x):
image

Suggested tweaks:
- move search box to display consistently beneath page title
- extend search box to fill full width of space available
- move "upload app" and "uninstall" buttons to the top right (this change is also consistent with (non-mobile) changes @zarvox and I have been discussing; to move the buttons to the top right aligned with the page title).

image

### PR DESCRIPTION
**Pull request by zarvox**
Various DOM/CSS changes related to improving layout and styles:

* we show the user profile picture on the account button
  * and in the equivalent place in the mobile menu
* we show the "apps" and "grains" icons in the mobile menu
* a few buttons move into the heading of the page they affect
* rows and sizing are cleaned thanks to flexbox

Fixes #2126.
Fixes #2139.

Desktop widths:

app-list-page

grain-list-page

app-details-page

Mobile sizes:

menu-with-icons

app-list-page-mobile

grain-list-page-mobile

app-details-page-mobile

### CODE REVIEW

**Comment by zarvox** [comment]
TODO before merge:

* get updated incognito icon from @neynah and add to the build and use it where expected
* make behavior reasonable in the face of non-square profile pictures

File: shell/client/accounts/login-buttons.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ Template.accountButtons.helpers({
     }
 
     // TODO(someday): obtain an incognito icon, then return it here.
```
Comment (paulproteus) [inline shell/client/accounts/login-buttons.js]: Does this commit allow you to remove this comment?

Comment (zarvox) [inline shell/client/accounts/login-buttons.js]: oops yes

**Comment by kentonv** [comment]
Cool, I've always wanted the profile picture to be shown there!

**Pull request merged**